Cswang/et platform

ttw225 opened the pull request:
endpoints: user, amis, address, power_info, login, participant, login, news

ttw225 committed:
feat: Initial commit

ttw225 committed:
feat(resource/bems.py): multithread uploading data with different nodes

ttw225 committed:
feat(bems.py): platform return address for each uploader

login and return address

ttw225 committed:
fix:

ttw225 committed:
docs(.gitignore):

ttw225 committed:
style(logging.py): change datetime format

ttw225 committed:
docs(api.apib): API Document

include Restful and Web Socket

ttw225 committed:
docs(Pipfile): add flask-socketio

ttw225 committed:
docs(api doc): Add Makefile to Generate Right api.html

ttw225 committed:
docs(API doc and travis.yml): update API Doc and DB schema and init travis.yml

ttw225 committed:
docs(README): add Travis CI icon to README

ttw225 committed:
style: Formatting

ttw225 committed:
style(Makefile):

ttw225 committed:
docs(travis.yml): add "Notifications to Slack"

ttw225 committed:
docs(travis.yml): add notify template

ttw225 committed:
docs(pt): modify folder's name: platform -> pt

ttw225 committed:
docs(Makefile): add clean

ttw225 committed:
docs(travis.yml): add CI test script

ttw225 committed:
docs(README): Travis Status icon

ttw225 committed:
docs(Pipfile): Separate the development kit

ttw225 committed:
style(Comply with pylint review):

ttw225 committed:
feat(endpoints): Design the API according to the new architecture.

ttw225 committed:
feat(user): add resource: get user

ttw225 committed:
feat(endpoints): Add API : Get Amis

ttw225 committed:
feat(endpoints): change a way to create relationship between tables

relation create by child

ttw225 committed:
feat(endpoints' models): create all table models and test for insert datas

ttw225 committed:
feat(endpoints and DB): endpoints: news, Electricity History Usage; DB: postgres

ttw225 committed:
feat(endpoints): api: login, change passwd, participants

ttw225 committed:
feat(add api endpoints):

ttw225 committed:
style(models): adjust format

Sirius207 commented on .gitignore:
```diff
@@ -102,3 +102,40 @@ venv.bak/
+
+# config file
+config.py
+
```
Suggest not ignoring config.py

Instead you can put a runnable sample config file with apparent dev value setup,  like DB_USER: 'dev_xx_user', 'dev_pw'

Sirius207 replied:
Besides, Travis CI can not run with a runnable config.py if your application need this file.

ttw225 replied:
Fix in e26cb24.

Sirius207 commented on .gitignore:
```diff
@@ -102,3 +102,40 @@ venv.bak/
+
+# testing folder
+tests/
+
```
Why do you ignore tests directory?

ttw225 replied:
Fix in 512f49b.
No testing files yet.

Sirius207 commented on .travis.yml:
```diff
@@ -0,0 +1,18 @@
+  slack:
+    rooms:
+      - secure: VJsOmi1mzr/QcVxZu32L0GAIFHvRn05BIxO9R0c/9lRbNNwUd3345/TjE9tJis5wvkmUQnCWqxNSn4MTiy4ETe40s0cIdrMexS0JG2s58MGMq+L2meKTtZxGiMLQGFDbkqvU6j6U7xlCc1KkXr46YJCP9ZmER04n6TbmpqoISJX6h/2IQz0MY+QR9f6loAfbeNw4prEYviwpJ5gw59w4x+jHUQqGu6KwhwsT7vclqYPyRmfBBsxvzsz4QOQABuwp7MM0lSKvBIyqibMtLFU7v88QmVrUM6QF1GoYlgh02hG6BdF49NjC499bep4B8s59zs1LTj0tuZNqx4m/PT/jQQMNErYHzhqEMpm7q4+M1cJTKf4sOu/GeMkzp146siy ...
+    on_success: always  # change: send a notification when the build status changes.
```
You should not put secure value in .travis.yml.

Try to read it by repo setting

https://docs.travis-ci.com/user/environment-variables/#defining-variables-in-repository-settings

ttw225 replied:
This is an encrypted secure value, don't worry.
Setting in travis' setting page might be a nice way.

Sirius207 commented on Dockerfile:
```diff
@@ -0,0 +1,11 @@
+WORKDIR /app
+
+RUN pip install -r requirements.txt
+
```
suggest using pipenv

exmaple 

```
FROM python:3.7.3-slim

# Install Pipenv
RUN pip install pipenv==2018.11.26
ENV LC_ALL C.UTF-8
ENV LANG C.UTF-8

# Install dependencies
COPY Pipfile Pipfile
COPY Pipfile.lock Pipfile.lock

RUN pipenv install --deploy --system
RUN pip install gunicorn==19.9.0

COPY ./udc_api /udc_api
WORKDIR /udc_api

RUN useradd -ms /bin/bash udc
RUN chown -R udc:udc /udc_api
USER udc

EXPOSE 5000

CMD ["gunicorn", "-c", "gunicorn.ini", "wsgi:app"]
```

ttw225 replied:
Fix in 512f49b.
Thanks a lot!!

Sirius207 commented on README.md:
```diff
@@ -1 +1,29 @@
+    `docker run --name mypostgres -e POSTGRES_PASSWORD=password -v "$PWD/db":/var/lib/postgresql/data -p 5432:5432 -d postgres`
+
+### 連接方式
+
```
"連接方式"  --> "XXX連接方式"   more readable

ttw225 replied:
Fix in 2cd051b.

Sirius207 commented on README.md:
```diff
@@ -1 +1,29 @@
++ run with db data
+
+    `docker run --name mypostgres -e POSTGRES_PASSWORD=password -v "$PWD/db":/var/lib/postgresql/data -p 5432:5432 -d postgres`
+
```
You can put a docker-compose file to run docker instead pulling it.


```
version: '3'
services:
  postgres:
    # checking connection: $ docker exec -it  ET_DB_POSTGRE psql -U {POSTGRES_USER} -W {POSTGRES_DB}
    image: postgres:11-alpine
    container_name: ET_DB_POSTGRE
    volumes:
      - ${POSTGRES_DATA_DIR}:/var/lib/postgresql/dev-data
    expose:
      - 5432
    ports:
      - ${POSTGRES_PORT}:5432
    environment:
      POSTGRES_DB: dev_db
      POSTGRES_USER: dev_user
      POSTGRES_PASSWORD: dev191022
```

ttw225 replied:
Fix in 18c843b.

Sirius207 commented on pt/app.py:
```diff
@@ -0,0 +1,79 @@
+from dotenv import load_dotenv
+from config import db
+from endpoints import RESOURCES
+
```
Is your flak8 runnable?

It will show that you have to put a blank line between 3-rd packages & custom packages

ttw225 replied:
Fix files in 9fd5f4f.

Sirius207 commented on pt/app.py:
```diff
@@ -0,0 +1,79 @@
+        resp = make_response(resp)
+        resp.headers['Access-Control-Allow-Origin'] = '*'
+        resp.headers['Access-Control-Allow-Methods'] = 'GET,POST'
+        resp.headers['Access-Control-Allow-Headers'] = 'x-requested-with,content-type'
```
You just allow GET & POST, but I saw DELETE & PUT In api document. 

Is  that right?

ttw225 replied:
Fix in 924cca0.

Sirius207 commented on pt/app.py:
```diff
@@ -0,0 +1,79 @@
+    app.run(
+        host=app.config['HOST'],
+        port=app.config['PORT']
+        # debug='no',
```
Is your port read from environment variable in config file?

Because travis would not run application in common port.

ttw225 replied:
Fix in e26cb24.

Sirius207 commented on .pylintrc:
```diff
@@ -0,0 +1,12 @@
+# R0913: too-many-arguments
+# R0902: too-many-instance-attributes
+# R0801: duplicate-code
+disable=C0111, R0903, W1203, E0611, E0401, R0913, R0902, R0801
```
suggest not ignoring R0801, It helps you write scalable code.

ttw225 replied:
OK!
Fix in 530a660.

Sirius207 commented on pt/endpoints/address/model.py:
```diff
@@ -0,0 +1,118 @@
+    def __repr__(self):
+        return self.uuid
+
+
```
If lots of model include
```
 def add(self):
        db.session.add(self)
        db.session.commit()

    # pylint: disable=R0201
    def update(self):
        db.session.commit()
    # pylint: enable=R0201

    def delete(self):
        db.session.delete(self)
        db.session.commit()

    def __repr__(self):
        return self.uuid
```

You can just inherit it.

ttw225 replied:
Fix in ef778ac.

Sirius207 commented on pt/endpoints/address/model.py:
```diff
@@ -0,0 +1,118 @@
+        return ami
+    renew(time)
+    return AMI.query.filter_by(tag=token, time=time).first()
+
```
what would happen if you run  `renew(time)` but  ami is still None?

Sirius207 replied:
maybe you can rename the function from `address` to `get_address`

ttw225 replied:
Function name fix in 10e4965.

Sirius207 commented on pt/endpoints/address/model.py:
```diff
@@ -0,0 +1,118 @@
+
+def renew(time):
+    # Update Address
+    new_address = get_addresses(int(time.strftime("%s")), AMI.query.count())
```
What is the purpose of this line?

ttw225 replied:
Fix in a9def17.

Sirius207 commented on pt/endpoints/address/resource.py:
```diff
@@ -0,0 +1,57 @@
+        logging.info(
+            "[Get Address Request]\nUser name:%s\nUUID:%s\nIOTA Address:%s"
+            % (g_ami.name, g_ami.uuid, g_ami.address)
+        )
```
You can simply use 
```
f"[Get Address Request]\nUser name:{g_ami.name}\nUUID:{g_ami.uuid}\nIOTA Address:{g_ami.address}"
```

in python3.

ttw225 replied:
After running on a new server with Python≥3.5, all print codes will use this method.
Thanks for suggestion! : )

Sirius207 commented on pt/endpoints/address/resource.py:
```diff
@@ -0,0 +1,57 @@
+            amis.append(
+                {"id": ami.uuid, "name": ami.name, "description": ami.description}
+            )
+
```
More pythonic example. for reference. (But it is faster)

```
amis = [
{"id": ami.uuid, "name": ami.name, "description": ami.description}
for ami in AMI.query.filter_by(user_id=g.uuid).all()
]
```
https://blog.teamtreehouse.com/python-single-line-loops

ttw225 replied:
Fix in 78e4dea.
Thanks a lot for your information!!

Sirius207 commented on pt/endpoints/bid/model.py:
```diff
@@ -0,0 +1,66 @@
+        db.session.commit()
+
+    def __repr__(self):
+        return self.uuid
```
You can inherit it.

```
def add(self):
        db.session.add(self)
        db.session.commit()

    # pylint: disable=R0201
    def update(self):
        db.session.commit()
    # pylint: enable=R0201

    def delete(self):
        db.session.delete(self)
        db.session.commit()

    def __repr__(self):
        return self.uuid

```

ttw225 replied:
Fix in ef778ac.

Sirius207 commented on pt/endpoints/bid/resource.py:
```diff
@@ -0,0 +1,41 @@
+                    "achievement": message.achievement,
+                    "settlement": message.settlement,
+                }
+            )
```
Again, you can use more pythonic statement instead using append.

ttw225 replied:
Fix in 78e4dea.

Sirius207 commented on pt/endpoints/data/model.py:
```diff
@@ -0,0 +1,121 @@
+
+
+class Homepage(Data):
+    __tablename__ = 'homepage'
```
Suggest not naming a table 'homepage'

It is not readable for recognizing what is inside.

ttw225 replied:
Rename homepage -> demand

Sirius207 commented on pt/endpoints/data/model.py:
```diff
@@ -0,0 +1,121 @@
+    __tablename__ = 'wt'
+    uuid = db.Column(db.String(40), db.ForeignKey('data.uuid'), primary_key=True)
+    cluster = db.Column(db.Integer)
+    WindGridPower = db.Column(db.Float)
```
Why  `WindGridPower `  is camelcase?

ttw225 replied:
Fix in 924cca0.

Sirius207 commented on pt/endpoints/data/resource.py:
```diff
@@ -0,0 +1,53 @@
+            time = date.today()
+        logging.info(
+            "[Get Datas Request]\nUser Account:%s\nUUID:%s\n" % (g.account, g.uuid)
+        )
```
Again, you can use f"something {some_variable}"

ttw225 replied:
This will be fix when using newer server : )

Sirius207 commented on pt/endpoints/data/resource.py:
```diff
@@ -0,0 +1,53 @@
+
+
+class DatasResource(Resource):
+    # pylint: disable=R0201
```
What is the purpose of ` DatasResource(Resource)`?
(What kinds of Data?)

ttw225 replied:
Change into PowerDatasResource.
Fix in 924cca0.

Sirius207 commented on pt/endpoints/user/resource.py:
```diff
@@ -0,0 +1,90 @@
+                "eth_address": user.eth_address,
+            }
+        )
+        response.status_code = 200
```
If you do not set the status_code 200, will it return other status code?

ttw225 replied:
Fix in db7106c.
Thanks for your information!

Sirius207 commented on pt/endpoints/user/resource.py:
```diff
@@ -0,0 +1,90 @@
+        data = request.get_json()
+        if User.query.filter_by(account=data['account']).first():
+            user = User.query.filter_by(account=data['account']).first()
+            if check_password_hash(user.password, data['password']):
```
How about
```
user = User.query.filter_by(account=data['account']).first()
if user:
   xxxx
```

more readable 💀

ttw225 replied:
Fix in 512f49b.

Sirius207 commented on pt/plat.py:
```diff
@@ -0,0 +1,22 @@
+TARGETADDRESS = (
+    'WIKETTNX9NVKOGZMOSTXMHFXV9ICADYSVJEPVUPTQQOLEV9YWCSWVPTNF9ETUXEHNJIXZPQWDPDOZ9999'
+)
+
```
Maybe you can put this address in environment variable

ttw225 replied:
Fix in 530a660.

Sirius207 commented on pt/utils/utils.py:
```diff
@@ -0,0 +1,27 @@
+# from config import API_URI
+
+API_URI = 'http://node10.puyuma.org:14265'
+
```
You should put `API_URI`  in config file.

ttw225 replied:
Fix in 530a660.

Sirius207 commented on pt/utils/verify.py:
```diff
@@ -0,0 +1,13 @@
+import base64
+import requests
+from config import APP as app
+
```
You can put a blank line between  `import request` &  `from config import ...`

ttw225 replied:
Fix in 530a660.

Sirius207 commented on pt/utils/verify.py:
```diff
@@ -0,0 +1,13 @@
+    json_data = json.dumps(data).encode('utf-8')
+    hash_data = hashlib.sha256(json_data).hexdigest().encode()
+    base64_data = base64.b64encode(hash_data).decode()
+    return bool(base64_data == message['message']['value'])
```
Is it necessary to put a bool function here?

ttw225 replied:
Fix in 530a660.

ttw225 committed:
fix(models): inherited models from base_models

reduce redundancy

ttw225 committed:
fix(endpoints/address/model): rename function address to get_address

ttw225 committed:
fix(endpoints/address/model): remove wrong line in function re_new()

ttw225 committed:
fix(endpoints/resources): use python-single-line-loops

ttw225 committed:
fix(endpoints/resources): remove setting status code to 200

ttw225 committed:
fix(iota_tracker.py): remove plat.py and add iota_tracker.py

iota_tracker.py is used to download IOTA's data, for default running per minutes

ttw225 committed:
fix(iota_tracker): fix wrong blank lines

ttw225 committed:
fix(user/resource, Dockerfile, .gitignore): Dockerfile is impressive now!

ttw225 committed:
fix(.py): put a blank line between 3-rd packages & custom packages

ttw225 committed:
fix(data, data/model): rename data -> power data

ttw225 committed:
fix(endpoints/powerdata/model): homepage -> summary

ttw225 committed:
fix(powerdata/model): rename Summary -> Demand

ttw225 committed:
feat(docker-compose.yml): add db's docker compose file

ttw225 committed:
feat(config and .env.sample): add config file and .env.sample file

ttw225 committed:
feat(README): add docker-compose, IOTA Tracker and Notice

ttw225 committed:
fix(config/__init__.py): add environment for db's URL

ttw225 committed:
fix(config/__init__.py): fix bugs

ttw225 committed:
fix(endpoints/powerdata/resource): Fix wrong attributes

Sirius207 commented on pt/endpoints/powerdata/model.py:
```diff
@@ -0,0 +1,141 @@
+        self.cluster = data_struct['cluster']
+        # pylint: disable=C0103
+        self.WindGridPower = data_struct['WindGridPower']
+        # pylint: enable=C0103
```
Is it  `self.windgridpower` ?

ttw225 replied:
Yes. Fix in 7b7af45.

Sirius207 commented on pt/endpoints/submit/model.py:
```diff
@@ -0,0 +1,31 @@
+
+
+class Submit(db.Model, ETBaseMixin):
+    __tablename__ = 'submit'
```
How about BidSubmit?

ttw225 replied:
BidSubmit is better~
Fix in 7b7af45.

Sirius207 commented on pt/iota_tracker.py:
```diff
@@ -0,0 +1,86 @@
+                    print(decrypt_data)
+                    # insert into db
+                    if tag[:-1] == 'BEMS9HOMEPAGE9INFORMATION9':
+                        Demand.add(
```
How about extract these tag to a dcit?
Maintainer might not know the meaning of these tag.

like
```
tag_dict = {
'some_tag': 'BEMS9HOMEPAGE9INFORMATION9',
'other_function_tag': 'BEMS9ESS9DISPLAY9',
...
}
```

ttw225 replied:
Fix in 7b7af45.

Sirius207 commented on pt/utils/oauth.py:
```diff
@@ -0,0 +1,23 @@
+    if user:
+        g.uuid = user.uuid
+        g.account = user.account
+        # logging.info("User Login: %s" % g.current_user)
```
Is this line need to be remove?

ttw225 replied:
Yes.
Fix in 7b7af45.

ttw225 committed:
fix(iota_tracker): move tags into one dict

ttw225 committed:
fix(iota_tracker.py): add main check
